OOT framebuffer emulation

DaMarkov opened the pull request:
Changed GLideN64 so that it is compatible with OOT with framebufer emulation being turned on.
Another PR to the OOT project will be necessary to turn framebuffer emulation on.
This PR should be backwards compatible with OOT with framebuffer emulation turned off.

Commit:
Changed output directoy.
Directoy is now relative to the project and not the solution.
Needed to add GLidenN64 to different solution files.

Commit:
Also changed the output folder of the other projects.

Commit:
Changed GLideN64 so it works with framebuffer emulation enabled with OOT.

blawar commented on projects/msvc/GLideN64.vcxproj:
```diff
@@ -46,7 +46,7 @@
     <CodeAnalysisRuleAssemblies />
-    <OutDir>$(SolutionDir)..\..\..\external\$(Platform)\</OutDir>
+    <OutDir>$(MSBuildProjectDirectory)\..\..\..\external\$(Platform)\</OutDir>
     <IntDir>$(SolutionDir)build\$(Platform)\$(Configuration)\$(ProjectName)\</IntDir>
```
None of the project file changes appear relevant to the PR, and should be removed.

blawar commented on src/FrameBuffer.cpp:
```diff
@@ -677,7 +677,8 @@ void FrameBufferList::setBufferChanged(f32 _maxY)
 {
 	gDP.colorImage.changed = FALSE;
-	FrameBuffer * pBuffer = frameBufferList().findBuffer(*REG.VI_ORIGIN & 0xffffff);
+	//FrameBuffer * pBuffer = frameBufferList().findBuffer(*REG.VI_ORIGIN & 0xffffff);
```
The original Gliden64 behavior should be supported.  Use the #ifdef NATIVE pre processor flag to signify native or original mode.

blawar commented on src/FrameBuffer.h:
```diff
@@ -145,7 +145,7 @@ class FrameBufferList
 	static FrameBufferList & get();
 
-private:
+//private:
```
Why was this commented out?

blawar commented on src/native/Native.cpp:
```diff
@@ -7,13 +7,18 @@
+//#define START_HEIGHT 720
 
-static u64 g_width = START_WIDTH;
+#define START_WIDTH  320
```
Why was the start size set to 320 x 240?

blawar commented on src/native/Native.cpp:
```diff
@@ -97,7 +102,7 @@ extern "C"
         config.video.windowedHeight = g_height;
         dwnd().setWindowSize(g_width, g_height);
-        (*REG.VI_ORIGIN)++;
+        //(*REG.VI_ORIGIN)++;
```
Just delete this

blawar commented on src/native/Native.cpp:
```diff
@@ -106,6 +111,111 @@ void _CheckInterrupts() {
+typedef struct OSMesgQueue {
+    /* 0x00 */ OSThread* mtqueue;
+    /* 0x04 */ OSThread* fullqueue;
+    /* 0x08 */ s32 validCount;
```
Let’s remove this, and just include the original files directly.  Add the ooot include path to the project.

blawar commented on src/native/Native.cpp:
```diff
@@ -133,9 +243,10 @@ extern "C" {
+        config.frameBufferEmulation.enable = 0;//Frame buffer disabled, will be enabled by OOT
+        //config.frameBufferEmulation.aspect = Config::aAdjust;
         
+        config.generalEmulation.hacks |= hack_subscreen | hack_ZeldaMonochrome;
```
Did you verify these hacks weren’t already being applied? I thought these hacks were already applied through the rom name logic?

blawar reviewed (changes requested):
It looks great so far, thank you!  Only a few small tweaks needed.

blawar commented:
Also change the default fbe option back to on.  It’s in some config cpp file, where the defaults are set in the constructor.

dmitry-markovich committed:
Changed the project files back.

dmitry-markovich committed:
Used NATIVE compiler flag to make changes for OOT optional.

Commit:
Moved structs into Native.h.

Commit:
Cleanup and missed on change in FrameBuffer.cpp.

DaMarkov commented:
Changed as requested except for including the necessary structs from the OOT project.
When including gfx.h I also get a bunch of macros (like _SHIFTL) from the ultra64 header files which are already defined in GLideN64.
I moved the structs to Native.h.
